internal/ethereum/blocklistener.go

```diff
@@ -292,12 +292,15 @@ func (bl *blockListener) rebuildCanonicalChain() *list.Element {
 	var notifyPos *list.Element
 	for {
 		var bi *blockInfoJSONRPC
+		var reason ffcapi.ErrorReason
 		err := bl.c.retry.Do(bl.ctx, "rebuild listener canonical chain", func(attempt int) (retry bool, err error) {
-			bi, err = bl.c.getBlockInfoByNumber(bl.ctx, nextBlockNumber, false, "")
+			bi, reason, err = bl.c.getBlockInfoByNumber(bl.ctx, nextBlockNumber, false, "")
 			return true, err
```

Comment:
does this need to be `return reason != ffcapi.ErrorReasonNotFound, err` to avoid us retrying on `ErrorReasonNotFound`?

Reply:
ah yep, thanks for catching that.